### PR TITLE
Test cases and fix for issue #1781

### PR DESCRIPTION
**Pull request by estollnitz**
This change includes test cases that were failing because of issue #1781: Internal assertion failures when compiling certain select expressions for OpenGL target.

In VaryingAttributes.cpp, `CastVariablesToFloatAndOffset` was mutating the true and false branches of a select expression but neglecting to mutate the condition. And for ramp expressions, if either base or stride mutates into a float, a cast may be require to make their types match.

Also added a diagnostic message to one of the assertions that was failing.

### CODE REVIEW

File: src/VaryingAttributes.cpp [hunk 1]
```diff
@@ -601,7 +602,29 @@ class CastVariablesToFloatAndOffset : public IRMutator {
...
 
-        expr = Select::make(op->condition, mutated_true_value, mutated_false_value);
+        expr = Select::make(mutated_condition, mutated_true_value, mutated_false_value);
+    }
+
+    virtual void visit(const Ramp *op) {
+        Expr mutated_base = mutate(op->base);
+        Expr mutated_stride = mutate(op->stride);
+
+        // If either base or stride is a float, then make sure both are float
```
Comment (abadams) [inline src/VaryingAttributes.cpp]: Maybe just call

    match_types(mutated_base, mutated_stride);

Comment (abadams) [inline src/VaryingAttributes.cpp]: Maybe explicitly handling the case you expect is better... LGTM either way.

Comment (estollnitz) [inline src/VaryingAttributes.cpp]: We only want to cast in the case where this class has mutated an int to a float; any other type changes might introduce errors or mask other issues. Therefore, I'd rather not call match_types, which could make many other changes.

**Comment by shoaibkamil** [comment]
LGTM .  Thanks for fixing this!

**Pull request merged**